projects/media-viewer/src/lib/store/store-utils.ts

```diff
@@ -0,0 +1,56 @@
+import {Annotation} from '../annotations/annotation-set/annotation-view/annotation.model';
+// @dynamic
+export class StoreUtils {
+
+  static generatePageEntities(annotations): {[id: string]: Annotation[]} {
+    return annotations.reduce((h, obj) =>
+      Object.assign(h, { [obj.page]:( h[obj.page] || [] ).concat(obj) }), {});
+  }
+
+  static generateCommentsEntities(annotations): {[id: string]: Annotation[]} {
```

Comment:
I think the return type is { [id: string]: Comment }

Reply:
yeap good spot